enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,983 @@
+    // It isn't used as a default value when eipAllocations is not specified on the IngressController.
+    EIPAllocations []EIPAllocations `json:"eipAllocations"`
+}
+
+type EIPAllocations string
+
+```
+
+- The API field for `eipAllocations` field in the Ingress Controller CR needs to be set as follows
+  in the file [operator/v1/types_ingress.go](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go):
+```go
+// AWSLoadBalancerParameters provides configuration settings that are
+// specific to AWS load balancers.
+// +kubebuilder:validation:XValidation:rule="self.type != 'Classic' || !has(self.nlbParameters)",message="Network load balancer parameters are allowed only when load balancer type is NLB."
+// +kubebuilder:validation:XValidation:rule="self.type != 'NLB' || !has(self.classicLoadBalancer)",message="Classic load balancer parameters are allowed only when load balancer type is Classic."
+// +union
+type AWSLoadBalancerParameters struct {
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer. Present only if type is NLB.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"nlbParameters,omitempty"`	
+}
+
+
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+type AWSNetworkLoadBalancerParameters struct {
+    // You can assign Elastic IP addresses to the Network Load Balancer by adding the following annotation. 
+    // The number of Allocation IDs must match the number of subnets that are used for the load balancer.
+    // service.beta.kubernetes.io/aws-load-balancer-eip-allocations: eipalloc-xxxxxxxxxxxxxxxxx,eipalloc-yyyyyyyyyyyyyyyyy
+    // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+    // https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/guide/service/annotations/#eip-allocations
+    // +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+    // The EIPs in the cluster config are only used when the operator creates the default IngressController.
+    // It isn't used as a default value when eipAllocations is not specified on the IngressController.
+    EIPAllocations []configv1.EIPAllocations `json:"eipAllocations"`
+}
+
+```
+
+### Workflow Description
+
+Pre-requisites: One must allocate an Elastic IP address from [Amazon's pool of public IPv4 addresses](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-eips-allocating),
+or from a custom IP address pool that you have brought to your AWS account. For more information about bringing your own IP address range to your 
+AWS account, see [Bring your own IP addresses (BYOIP) in Amazon EC2](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-byoip.html).
```

Comment:
*nit* is this period supposed to be there?

```suggestion
AWS account, see [Bring your own IP addresses (BYOIP) in Amazon EC2](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-byoip.html)
```

Reply:
no